opentelemetry-exporter-otlp-proto-grpc: bump grpcio min version

### Describe your environment

Cross-reference: https://github.com/grpc/grpc/issues/38790

OS: ubuntu-latest
Python version: 3.13 specifically
SDK version: 1.30.0
API version: 1.30.0


### What happened?

The lowest allowed version for `grpcio` cannot be built on Python 3.13.


### Steps to Reproduce

```
# fresh project
uv add opentelemetry-exporter-otlp-proto-grpc==1.30.0
uv sync --resolution lowest
```

### Expected Result

venv installed

### Actual Result

```py
...
src/python/grpcio/grpc/_cython/cygrpc.cpp: In function ‘void
      __Pyx_init_assertions_enabled()’:
      src/python/grpcio/grpc/_cython/cygrpc.cpp:6354:39: error:
      ‘_PyInterpreterState_GetConfig’ was not declared in this scope; did you
      mean ‘PyInterpreterState_GetID’?
       6354 |     __pyx_assertions_enabled_flag = !
      _PyInterpreterState_GetConfig(__Pyx_PyThreadState_Current->interp)->optimization_level;
            |
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            |                                       PyInterpreterState_GetID
...
      distutils.errors.CompileError: command '/usr/bin/c++' failed with exit
      code 1
...
        `opentelemetry-exporter-otlp-proto-grpc` (v1.30.0) which depends on
        `grpcio`
```

### Additional context

I don't know which project should solve this, gRPC or OTEL.

### Would you like to implement a fix?

None

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#4373](https://github.com/open-telemetry/opentelemetry-python/pull/4373))
 - opentelemetry-sdk: fix OTLP exporting of Histograms with explicit buckets advisory
   ([#4434](https://github.com/open-telemetry/opentelemetry-python/pull/4434))
+- opentelemetry-exporter-otlp-proto-grpc: better dependency version range for Python 3.13
+  ([#4444](https://github.com/open-telemetry/opentelemetry-python/pull/4444))
+- opentelemetry-exporter-opencensus: better dependency version range for Python 3.13
+  ([#4444](https://github.com/open-telemetry/opentelemetry-python/pull/4444))
 
 ## Version 1.30.0/0.51b0 (2025-02-03)
 

--- a/exporter/opentelemetry-exporter-opencensus/pyproject.toml
+++ b/exporter/opentelemetry-exporter-opencensus/pyproject.toml
@@ -29,7 +29,8 @@ classifiers = [
   "Typing :: Typed",
 ]
 dependencies = [
-  "grpcio >= 1.0.0, < 2.0.0",
+  "grpcio >= 1.63.2, < 2.0.0; python_version < '3.13'",
+  "grpcio >= 1.66.2, < 2.0.0; python_version >= '3.13'",
   "opencensus-proto >= 0.1.0, < 1.0.0",
   "opentelemetry-api >= 1.31.0.dev",
   "opentelemetry-sdk >= 1.15",

--- a/exporter/opentelemetry-exporter-otlp-proto-grpc/pyproject.toml
+++ b/exporter/opentelemetry-exporter-otlp-proto-grpc/pyproject.toml
@@ -30,7 +30,8 @@ classifiers = [
 dependencies = [
   "Deprecated >= 1.2.6",
   "googleapis-common-protos ~= 1.52",
-  "grpcio >= 1.63.2, < 2.0.0",
+  "grpcio >= 1.63.2, < 2.0.0; python_version < '3.13'",
+  "grpcio >= 1.66.2, < 2.0.0; python_version >= '3.13'",
   "opentelemetry-api ~= 1.15",
   "opentelemetry-proto == 1.31.0.dev",
   "opentelemetry-sdk ~= 1.31.0.dev",

